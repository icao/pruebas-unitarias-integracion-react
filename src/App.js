import React from 'react'
import './App.css'

function App() {
  return (
    <div className='App'>
      <header className='App-header'>
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className='App-link'
          href='https://github.com/icao/pruebas-unitarias-integracion-react'
          target='_blank'
          rel='noopener noreferrer'
        >
          Repositorio de pruebas unitarias e integracion en React.js
        </a>
      </header>
    </div>
  )
}

export default App
