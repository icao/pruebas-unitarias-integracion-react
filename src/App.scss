@import './scss/Palette';
@import './scss/Utils';
@import './scss/Fonts';

.app {
  &-header {
    width: 100%;
    background-color: $color-primary;
    position: sticky;
    top: 0;
    padding: 0.5em;
    color: $color-white;
  }
  &-main {
    padding: 0.7em;
    padding-bottom: 3em;
    background-color: $color-primary;
    min-height: 100vh;
    height: auto;
    color: $color-white;
    @extend %full-center;
  }
  &-footer {
    width: 100%;
    position: fixed;
    bottom: 0;
    padding: 0.5em;
    font-size: calc(8px + 1vmin);
    text-align: center;

    &__bg-link {
      display: inline-block;
      background-color: rgba(255, 255, 255, 0.2);
      border-radius: 5px;
      padding: 2px 15px;
    }
  }
}

.title {
  font-size: calc(20px + 2vmin);
  text-align: center;
  margin: 0;
}

.link {
  color: #00d9ff;
}

.test-method {
  color: #00d9ff;
}
