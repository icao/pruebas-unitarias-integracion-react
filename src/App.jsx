import React from 'react'
import './App.scss'
import MiPrimerComponenteTest from './components/01-mi-primer-test/MiPrimerComponenteTest'

function App() {
  return (
    <div className='app'>
      <header className='app-header'>
        <h1 className='title'>
          Pruebas unitarias y de integraci√≥n en react.js
        </h1>
      </header>
      <main className='app-main'>
        <MiPrimerComponenteTest />
      </main>

      <footer className='app-footer'>
        <span className='app-footer__bg-link'>
          <a
            className='link'
            href='https://github.com/icao/pruebas-unitarias-integracion-react'
            target='_blank'
            rel='noopener noreferrer'
          >
            Repositorio de pruebas unitarias e integracion en React.js
          </a>
        </span>
      </footer>
    </div>
  )
}

export default App
