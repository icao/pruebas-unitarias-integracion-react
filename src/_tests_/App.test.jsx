import React from 'react'
import { render } from '@testing-library/react'
import App from '../App'

/**
 * Prueba con REACT Testing Library
 */
describe('Probando componente <App /> con React Testing Library', () => {
  test('App debe mostrar un titulo: Pruebas unitarias y de integración - React', () => {
    let titulo = 'Pruebas unitarias y de integración - React'
    const wrapper = render(<App title={titulo} />) // [1]
    expect(wrapper.getByText(titulo)).toBeInTheDocument() // [2]
  })
})

/**
 * Probando la renderización de un componente de React
 *
 * [1] - Primero tenemos que importar nuestro componente
 * import React from 'react'    - Importamo React, ya que hacemos uso de nuesto componente
 * import App from '../App'   - Importando el componente
 * import { render } from '@testing-library/react' - Importamos el método "render()" desde la "testing library"
 *
 * Guardamos en una variable lo que regrese el método render([componente])
 *
 * [2] - Harémos uso de la "testing library" de React para testear nuestro componente y obtener algunos valores.
 * Usamos el método getByText([texto_a_buscar]), que recibe como parámetro el texto que queremos buscar dentro del componente.
 *
 * Y con el método toBeInTheDocument() de Jest, comparamos si el texto(wrapper) coincide con alguno dentro del componente.
 *
 */

/**
 * IMPORTANTE: Para que Jest pueda hacer uso de este método getByText(), tenemos que configurar un archivo para que Jest pueda hacer uso de una libreria que necesita para acceder a nuestro DOM.
 *
 * En la carpeta "src", creamos un archivo de configuración que llamaremos: setupTest.js
 * E importaremos  a jest-dom de la siguiente manera:
 * import '@testing-library/jest-dom/extend-expect'
 *
 * Una vez hecho, requeriremos volver a levantar nuestras pruebas con yarn test, para cargar las configuraciones realizadas.
 *
 */

/**
 * Prueba con ENZYME
 */

describe('Probando componente <App /> con Enzyme', () => {
  test('App debe mostrar un titulo: Pruebas unitarias y de integración - React', () => {
    console.log('ENZYME')
  })
})
