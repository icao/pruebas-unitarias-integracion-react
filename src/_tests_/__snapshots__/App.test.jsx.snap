// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <App /> con Enzyme App debe mostrarse correctamente 1`] = `
<div
  className="app"
>
  <header
    className="app-header"
  >
    <h1
      className="title"
    >
      Pruebas unitarias y de integraci√≥n - React
    </h1>
    <h2
      className="subtitle"
    >
      Default Jest & Enzime
    </h2>
  </header>
  <main
    className="app-main"
  >
    <MiPrimerComponenteTest />
  </main>
  <footer
    className="app-footer"
  >
    <span
      className="app-footer__bg-link"
    >
      <a
        className="link"
        href="https://github.com/icao/pruebas-unitarias-integracion-react"
        rel="noopener noreferrer"
        target="_blank"
      >
        Repositorio de pruebas unitarias e integracion en React.js
      </a>
    </span>
  </footer>
</div>
`;
